File: Leetcode/Diet_Plan_Performance.cpp
class Solution {
public:
    int dietPlanPerformance(vector<int>& calories, int k, int lower, int upper) {

        int total =0;
        int runner = 0;
        for (int i =0 ; i< k;i+=1)
            runner += calories[i];
        if (runner > upper)
            total += 1;
        else if (runner < lower)
            total -=1;
        for (int i =k; i < calories.size();i+=1)
        {
            runner -= calories[i-k];
            runner+= calories[i];
            if (runner > upper)
            total += 1;
        else if (runner < lower)
            total -=1;
        }
        return total;
    }
};

File: Leetcode/Diet_Plan_Performance.java
class Solution {
    public int dietPlanPerformance(int[] calories, int k, int lower, int upper) {
       
            int tot =0 , curr =0;
            for (int i=0; i< k;i+=1)
            {
                curr += calories[i];
            }
            if (curr > upper)
                tot +=1;
            else if (curr < lower)
                tot-=1;
        
            for (int i = k; i < calories.length;i+=1)
            {
                curr -= calories[i-k];
                curr += calories[i];
                if (curr > upper)
                    tot +=1;
                else if (curr < lower)
                    tot-=1;                
            }
        return tot;
    }
}
